core/vdbe/mod.rs

```diff
@@ -1699,10 +1707,26 @@ fn exec_char(values: Vec<OwnedValue>) -> OwnedValue {
     OwnedValue::Text(Rc::new(result))
 }
 
-// Implements LIKE pattern matching.
-fn exec_like(pattern: &str, text: &str) -> bool {
-    let re = Regex::new(&pattern.replace('%', ".*").replace('_', ".").to_string()).unwrap();
-    re.is_match(text)
+fn construct_like_regex(pattern: &str) -> Regex {
+    Regex::new(&pattern.replace('%', ".*").replace('_', ".").to_string()).unwrap()
+}
+
+// Implements LIKE pattern matching. Caches the constructed regex if a cache is provided
+fn exec_like(regex_cache: Option<&mut HashMap<String, Regex>>, pattern: &str, text: &str) -> bool {
+    if let Some(cache) = regex_cache {
+        match cache.get(pattern) {
+            Some(re) => re.is_match(text),
+            None => {
+                let re = construct_like_regex(pattern);
+                let res = re.is_match(text);
+                cache.insert(pattern.to_string(), re);
```

Comment:
I wonder if there's logic to remove old keys inside the cache 🤔 . Wouldn't the size of the hashMap continuously grow without shrinking?

Reply:
Since the cache is part of `ProgramState`, it only lives for the duration of a single call to `Connection::execute` or the lifetime of a `Statement`, so it's got a short life, right? Or am I missing something about how long a `Statement` can live for?

Reply:
Oh, missed that it's hold for a short period. Then I wonder if it's worth to cache short living regexes? Let me first look into sqlite3 implementations if exists.

Reply:
Sounds good. I can work on doing some benchmarking of with-cache and without-cache versions to see how much of a difference it makes on queries over a lot of rows.

Reply:
Thank you for the great detail! 🙇